src/pages/why-isnt-x-a-hook/index.de.md

```diff
@@ -0,0 +1,237 @@
+  return (
+    <ChatLayout width={width}>
+      <FriendStatus isOnline={isOnline} />
+      {isTyping && 'Schreibt...'}
+    </ChatLayout>
+  );
+}
+```
+
+Wann rendert es neu?
+
+Wenn jeder `useBailout()`-Aufruf die Möglichkeit hätte ein Update auszulassen, dann wären die Updates von `userWindowWidth()` durch `useFriendStatus()` geblockt und andersherum. **Diese Hooks würden sich gegen blockieren**.
+
+Aber wenn `useBailout()` erst dann respektiert werden würde, wenn *alle* Aufrufe aus einer Komponente das Blocken des Updates akzeptieren würden, dann würde unser `ChatThread` nicht funktionieren und keine Änderungen an dem `isTyping` Props updaten.
+
+Noch schlimmer ist, dass dies dazu führt das **alle neuen Hooks für `ChatThread` kaputt gehen würden, wenn sie nicht *auch* `useBailout()` aufrufen würden**. Andererseits können sie sich nicht gegen den Bailout in `useWindowWith()` und `useFriendStatus()` "wehren".
+
+**Fazit:** 🔴 `useBailout()` zerstört eine gemeinsame Benutzung. Wenn man es zu den Hooks hinzufügt, macht es andere State-Updates in anderen Hooks kaputt. Wir wollen, dass die APIs [funktionssicher](/optimized-for-change/) sind und dieses Verhalten ist ziemlich genau das Gegenteil.
+
+### Debugging
+
+Wie beinflusst ein Hook wie `useBailout()` das Debugging?
+
+Dafür nutzen wir das gleiche Beispiel:
+
+```js
+function ChatThread({ friendID, isTyping }) {
+  const width = useWindowWidth();
+  const isOnline = useFriendStatus(friendID);
+  return (
+    <ChatLayout width={width}>
+      <FriendStatus isOnline={isOnline} />
+      {isTyping && 'Schreibt...'}
+    </ChatLayout>
+  );
+}
+```
+
+Wir sagen, dass `Schreibt...` nicht dann angezeigt wird wenn wir es erwartet haben, obwohl irgendwie viele Ebenen darüber sich das Prop sich ändert. Wie debuggen wir das?
+
+**Normalerweise kann man mit voller Überzeugung sagen, dass man bei React einfach *nachschaut*.** Wenn `ChatThread` keinen neuen `isTyping` Wert bekommt, können wir die Komponente, die `<ChatThread isTyping={myVar} />` rendert, öffnen und `myVar` checken. Auf einer Ebene werden wir entweder ein verbuggtes `shouldComponentUpdate()` oder einen falschen `isTyping` Wert finden. Ein Blick auf jede Komponente in dieser Kette reicht normalerweise aus, um die Ursache des Problems zu finden.
+
+Aber wenn jedoch dieser `useBailout()` Hook existieren würde, würde man nie den wahren Grund warum ein Update übersprungen wurde finden bis man *jeden einzelnen erstellen Hook* (bis in die Tiefe), der von unserem `ChatThread` und Komponenten in seiner Kette verwendet wird, überprüft hätte. Weil jede einzelne Parent-Komponente *auch* selbsterstellte Hooks benutzen kann, [skaliert](/the-bug-o-notation/) diese Suche schrecklich.
+
+Das ist so, als würde man nach einem Schraubenzieher in einer Kiste voller Schubladen suchen und jede Schublade würde eine Menge kleinerer Kisten voller Schubladen beinhalten und man weiß nicht wie Tief diese Höhle weiter geht.
```

Comment:
```suggestion
Das ist so, als würde man nach einem Schraubenzieher in einer Kiste voller Schubladen suchen und jede Schublade würde eine Menge kleinerer Kisten voller Schubladen beinhalten und man weiß nicht wie tief diese Höhle weiter geht.
```
"Tief" should be lowercase because it is an adjective.